File: Core/CMS.Domain/Entities/CompanyMaster/MasterCompany.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CMS.Domain.Entities.CompanyMaster
{
    public class MasterCompany
    {
        [Key]
        public int ValueId { get; set; }
        public string CompanyName { get; set; }

        public string PocName { get; set; }

        
        public bool CompanyStatus { get; set; } = true;

        public long PocContactNumber { get; set; }

        public string PocEmailId { get; set; }


        //Company Contact details

        public string CompanyAddressLine1 { get; set; }
        public string CompanyAddressLine2 { get; set; }
        public string CompanyAddressLine3 { get; set; }

        public int Zipcode { get; set; }
        public long CompanyContactNo { get; set; }

        public string CompanyEmailId { get; set; }

        public string CompanyWebsiteUrl { get; set; }


        //Company Bank/Other Details

        public string CompanyBankName { get; set; }

        public long GSTno { get; set; }

        public long BankAccNo { get; set; }

        public long MSMERegistrationNo { get; set; }

        public string IFSCCode { get; set; }

        public string PanNo { get; set; }


        //for soft delete 
        public bool IsDeleted { get; set; } = false;

        //
        public ListOfCountries country;
        public int CountryId { get; set; }

       
        public ListOfStates state;
        public int StateId { get; set; }
   

        public ListofCity city;
        public int CityId { get; set; }

      


    }
}

File: Frontend/CMS/src/app/models/master-company.ts
export class MasterCompany {
    valueId?:number;
    companyName?:string;
    pocName?:string;
    companyStatus?:boolean;
    pocContactNumber?:number;
    pocEmailId?:string;
    companyAddressLine1?:string;
    companyAddressLine2?:string;
    companyAddressLine3?:string;
    zipcode?:number;
    companyContactNo?:number;
    companyEmailId?:string;
    companyWebsiteUrl?:string;
    companyBankName?:string;
    gSTno?:number;
    bankAccNo?:number;
    mSMERegistrationNo?:number;
    iFSCCode?:string;
    panNo?:string;
    countryId?:number;
    stateId?:number;
    cityId?:number;
}

export class CompanyMasterDto{
    //companies:MasterCompany[];
    valueId:number;
    companyName:string;
    companyLocation:string;
    status:boolean;
    TotalRecords:number;

    
    constructor(valueId:number,companyName:string,companyLocation:string,status:boolean,TotalRecords:number) {
    this.valueId=valueId;
    this.companyName=companyName;
    this.companyLocation=companyLocation;
    this.status=status;
    this.TotalRecords=TotalRecords;      
    }
}

export class AddCompanyDto{
    //without constructor initializing values using nullable type
    companyName?:string | null;
    valueId?:number | null;
    pompanyName?:string | null;
    pocName?:string | null;
    companyStatus?:boolean | null;
    pocContactNumber?:number | null;
    pocEmailId?:string | null;
    companyAddressLine1?:string | null;
    companyAddressLine2?:string | null;
    companyAddressLine3?:string | null;
    zipcode?:number | null;
    companyContactNo?:number | null;
    companyEmailId?:string | null;
    companyWebsiteUrl?:string | null;
    companyBankName?:string | null;
    gSTno?:number | null;
    bankAccNo?:number | null;
    mSMERegistrationNo?:number | null;
    iFSCCode?:string | null;
    PanNo?:string | null;
    countryId?:number | null;
    stateId?:number | null;
    cityId?:number | null;
}

export class CompanyListResponse {
    companies: CompanyMasterDto[] = [];
    totalCount: number = 0;
  }
  
